lib/facility.rb

```diff
@@ -1,14 +1,141 @@
+  end
+
+  def co_short_address_input?(data)
+    data[:address_li] && data[:city] && data[:state] && data[:zip]
+  end
+
+  def co_short_address_output(data)
+    "#{data[:address_li]}, #{data[:city]}, #{data[:state]} #{data[:zip]}"
+  end
+
+  def ny_address_input?(data)
+    data[:street_address_line_1] && data[:city] && data[:state] && data[:zip_code]
+  end
+
+  def ny_address_output(data)
+    "#{data[:street_address_line_1]}, #{data[:city]}, #{data[:state]} #{data[:zip_code]}"
+  end
+
+  def mo_address_input?(data)
+    data[:address1] && data[:city] && data[:state] && data[:zipcode]
+  end
+
+  def mo_address_output(data)
+    "#{data[:address1]}, #{data[:city]}, #{data[:state]} #{data[:zipcode]}"
+  end
+
+  def parse_address(data)
+    if data[:address]
+      data[:address] 
+    elsif co_address_input?(data)
+      co_address_output(data)
+    elsif co_short_address_input?(data)
+      co_short_address_output(data)
+    elsif ny_address_input?(data)
+      ny_address_output(data)
+    elsif mo_address_input?(data)
+      mo_address_output(data)
+    end
+  end
+    
+  def parse_phone(data)
+    if data[:phone]
+      data[:phone]
+    else data[:public_phone_number]
+      data[:public_phone_number]
+    end
+  end
+  
+  def parse_services(data)
+    if data[:services_p]
+      data[:services_p]
+    else
+      []
+    end
+  end
+
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration') == true && vehicle.antique? == true
+      vehicle.registration_date = Date.today
+      @collected_fees += 25
+      @registered_vehicles << vehicle
+      vehicle.plate_type = :antique
+    elsif @services.include?('Vehicle Registration') == true && vehicle.electric_vehicle? == true
+      vehicle.registration_date = Date.today
+      @collected_fees += 200
+      @registered_vehicles << vehicle
+      vehicle.plate_type = :ev
+    elsif @services.include?('Vehicle Registration') == true
+      vehicle.registration_date = Date.today
+      @collected_fees += 100
+      @registered_vehicles << vehicle
+      vehicle.plate_type = :regular
+    else @services.include?('Vehicle Registration') == false
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') == true && registrant.permit == true && registrant.age >= 16
+      return registrant.license_data[:written] = true
+    else
+      false
+    end
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?('Road Test') == true && registrant.license_data[:written] = true
+      return registrant.license_data[:license] = true
+    else
+      false
+    end
+  end
+    
+  def renew_drivers_license(registrant)
+    if @services.include?('Renew License') == true && registrant.license_data[:license] == true
+      return registrant.license_data[:renewed] = true
+    else
+      false
+    end
+  end
 end
+
+
```

Comment:
For `administer_written_test`, `administer_road_test`, `renew_drivers_license` there is not need for the explicit `return` if you are using the else. Removing the explicit `return` would produce the same result. You would only need it if you did not have the else.

```ruby
def administer_written_test(registrant)
  if @services.include?('Written Test') && registrant.permit && registrant.age >= 16
    registrant.license_data[:written] = true
  else
    false
  end
end
```

And this is the same:

```ruby
def administer_written_test(registrant)
  if @services.include?('Written Test') && registrant.permit && registrant.age >= 16
    return registrant.license_data[:written] = true
  end

  false
end
```